src/main/java/games/strategy/triplea/ui/menubar/LobbyMenu.java

```diff
@@ -298,38 +303,38 @@ private long requestTimespanSupplication() {
             JOptionPane.OK_CANCEL_OPTION, JOptionPane.QUESTION_MESSAGE, null, timeUnits.toArray(),
             timeUnits.toArray()[3]);
     if (result < 0) {
-      return -1;
+      return Duration.ofMillis(-1);
```

Comment:
I think the original author of this code intended -1 to mean "the user canceled the operation."  It was probably a mistake to treat that value as a -1 ms duration to be added to the current time.  At a minimum, the caller of this method probably should have checked for -1 and aborted the ban operation.

With today's code, specifying an instant before the current time will cause any existing ban to be removed.  So, if the admin cancels out of one of the dialogs, it will have the unintended consequence of actually removing any ban currently in place.

This goes back to what we discussed in your other PR about potentially returning an `Optional` from this method.  However, given what @DanVanAtta suggested below, that begs the question of how you represent forever.  If you continue to use `null` (which you could do), you would end up with a return type of `@Nullable Optional<TemporalAmount>`.  That just looks weird and seems very confusing.

I honestly don't know the best way you should represent these three possible value types (abort ban, ban forever, and ban for finite duration).  I'll throw out what I suggested in your other PR, but I'm still not sure it's ideal:

* Use `Optional<TemporalAmount>` as the return type.
* An empty `Optional` indicates the user aborted the ban, and no further action should be taken.
* A non-empty `Optional` indicates the user wants to proceed with the ban; the `Optional` value represents the duration of the ban.
* Use a special `TemporalAmount` value to represent forever.  For example, start by defining an appropriate magic constant (note that the actual duration could be anything because you're going to use reference equality below to test for this value):

```java
private static final TemporalAmount FOREVER = Duration.ofSeconds(Long.MAX_VALUE);
```

Obviously, you'd return this instead of `null` from `requestTimestampSupplication()`:

```java
if (selectedTimeUnit.equals("Forever")) {
  return Optional.of(FOREVER);
}
```

And then test specifically for this magic value **using reference equality** in `toDate()`:

```java
return amount != FOREVER
    ? Date.from(LocalDateTime.ofInstant(Instant.now(), ZoneOffset.UTC).plus(amount).toInstant(ZoneOffset.UTC))
    : null;
```

Note that this solution conflicts with @DanVanAtta's suggestion below:

>  The semantic meaning of a variable changing based only on its value adds a decent level of complexity, makes it harder to reason about the logic.

But I'm hoping naming the "special value" `FOREVER` instead of using `Long.MAX_VALUE` makes it easier to reason about. :smile:

Reply:
Interesting context, in this case I would recommend creating a data object to capture the return data. It would likely have two fields, one an enum to indicate result and a second to indicate the value in case the ban value is in the future.

Reply:
@ssoloff This reminded me of something, there is actually a chrono unit which represents the concept of "forever": https://docs.oracle.com/javase/9/docs/api/java/time/temporal/ChronoUnit.html#FOREVER

Will use that instead!

Reply:
Internally it uses Long.MAX_VALUE

Reply:
Actually canceling removed the mac even before I fiddled with the time-code, with the exception, that the a value in the past was inserted into the db which gets removed after the next person with this mac joins
Those old classes are full of bugs...
Is fixed now.

Reply:
:+1: Thanks for the heads-up on `ChronoUnit#FOREVER`.

> Actually canceling removed the mac even before I fiddled with the time-code, with the exception

Yes, it was a legacy bug; definitely not introduced in this PR.  Sorry, if I didn't make that clear.